src/storage/temporary_file_manager.cpp

```diff
@@ -269,6 +307,32 @@ bool TemporaryFileManager::HasTemporaryBuffer(block_id_t block_id) {
 	return used_blocks.find(block_id) != used_blocks.end();
 }
 
+idx_t TemporaryFileManager::GetTotalUsedSpaceInBytes() {
+	return size_on_disk.load();
+}
+
+void TemporaryFileManager::IncreaseSizeOnDisk(idx_t bytes) {
+	auto &config = DBConfig::GetConfig(db);
+	auto max_swap_space = config.options.maximum_swap_space;
```

Comment:
Do we want to be able to adjust this at run-time while the temporary directory is in use? Currently this implementation allows for it - but similarly to the temp directory setting we could also say that this has to be specified prior to creating the temp directory.

Reply:
If we want to allow this, then we need to read from `config.options.maximum_swap_space` here at every iteration to make sure it has not changed in the meantime.

Reply:
And also, if we want to allow for this, what do we do if the setting is lowered while the limit is already exceeded?

Reply:
Can we add this as a test using e.g. a temporary table with a low memory limit.